backends-clickhouse/src/main/scala/org/apache/gluten/execution/CHHashAggregateExecTransformer.scala

```diff
@@ -349,13 +349,18 @@ case class CHHashAggregateExecTransformer(
               (makeStructTypeSingleOne(child.dataType, child.nullable), child.nullable)
             case covar: Covariance =>
               var fields = Seq[(DataType, Boolean)]()
-              fields = fields :+ (covar.left.dataType, covar.left.nullable)
-              fields = fields :+ (covar.right.dataType, covar.right.nullable)
+              // when COVAR_SAMP's input is non-nullable , due to its output maybe null in spark,
+              // backend parser will cast input type to nullable for intermediate agg,
+              // While final stage still use input type no changed.
+              // It causes type miss match between partial and final AggregationFunction
+              // so directly assign attr.nullable here, which should always be nullable
+              fields = fields :+ (covar.left.dataType, attr.nullable)
+              fields = fields :+ (covar.right.dataType, attr.nullable)
```

Comment:
is there any case?

Reply:
I am concerned about this change. If the nullability of input is different from result, it could cause an exception when CH parses the intemediate aggregate result in the final stage. How could this  avoid ?

Reply:
Please refer to the newly added test case "covar_samp non-nullable input", which use a temp inline table  and all columns are non-nullable.Before this fix, it will fail.

Reply:
Is it that the intermediate result will be nullable any way in `CH` ?